--- ./src/evdev.c.orig	2026-02-12 17:25:46.000000000 -0600
+++ ./src/evdev.c	2026-02-13 16:30:08.507757486 -0600
@@ -1019,7 +1019,7 @@
 
 	/* if we have a current libinput_dispatch() snapshot, compare our
 	 * event time with the one from the snapshot. If we have more than
-	 * 10ms delay, complain about it. This catches delays in processing
+	 * 75ms delay, complain about it. This catches delays in processing
 	 * where there is no steady event flow and thus SYN_DROPPED may not
 	 * get hit by the kernel despite us being too slow.
 	 */
@@ -1028,7 +1028,7 @@
 		return;
 
 	tdelta = usec_to_millis(usec_delta(libinput->dispatch_time, eventtime));
-	if (tdelta > 20) {
+	if (tdelta > 75) {
 		evdev_log_bug_client_ratelimit(
 			device,
 			&device->delay_warning_limit,
