--- ./src/evdev.c.orig	2025-07-31 00:26:54.000000000 -0500
+++ ./src/evdev.c	2025-07-31 12:24:41.391982278 -0500
@@ -1018,7 +1018,7 @@
 
 	/* if we have a current libinput_dispatch() snapshot, compare our
 	 * event time with the one from the snapshot. If we have more than
-	 * 10ms delay, complain about it. This catches delays in processing
+	 * 75ms delay, complain about it. This catches delays in processing
 	 * where there is no steady event flow and thus SYN_DROPPED may not
 	 * get hit by the kernel despite us being too slow.
 	 */
@@ -1026,7 +1026,7 @@
 		return;
 
 	tdelta = us2ms(libinput->dispatch_time - eventtime);
-	if (tdelta > 20) {
+	if (tdelta > 75) {
 		evdev_log_bug_client_ratelimit(
 			device,
 			&device->delay_warning_limit,
